### DISCUSSION TITLE
Is there any paid version of yfinance?

### DISCUSSION THREAD

**Discussion by dushyant-chauhan** [ranaroussi/yfinance#1886] [Yahoo's data]
I am new to downloading NSE stocks data for personal use.

I started with yfinance and was able to download data of 1K of stocks in a day for few months. But now it seems I have been blacklisted or rated down by yfinance as I can get data of only 10 or 20 stocks.

Is there any way I can get myself removed from blacklist or even if there is any reasonable paid version of yfinance, I am okay with that?

**Comment by yiwenxu6** [comment]
@ValueRaider EDIT: **RapidAPI links are dead, don't click, just in case compromised.**

---

[RapidAPI](https://rapidapi.com/apidojo/api/yahoo-finance1 ) is a paid version of yfinance. However, they also have a limit, which may disappoint you (see [https://rapidapi.com/apidojo/api/yahoo-finance1/pricing](https://rapidapi.com/apidojo/api/yahoo-finance1/pricing) for the price). But if you are really serious about algorithmic trading using real money, I would recommend you use [Alpha Vantage](https://www.alphavantage.co/) instead of yfinance.

**Comment by dushyant-chauhan** [comment]
Thanks yiwenxu6 for the reply.

I tried looking for NSE (National Stock Exchange - India) data in Alpha Vantage but sadly it is not there.

Let me just quickly check for RapidAPI for the NSE data and codes.

**Comment by koradiyakaushal** [comment]
@dushyant-chauhan you can use angle broking API, they provide free historical data with their smartapi, you will have to open an account to generate an API key.
